website/api/controllers/entrance/signup.js

```diff
@@ -100,46 +104,46 @@ the account verification message.)`,
-    let cloudProvisionerResponse = await sails.helpers.http.post(
+    let cloudProvisionerResponseData = await sails.helpers.http.post(
       'https://sandbox.fleetdm.com/new',
       { // Request body
         'name': firstName + ' ' + lastName,
         'email': newEmailAddress,
-        'password': fleetSandboxDemoKey,
+        'password': fleetSandboxDemoKey, //« this provisioner API was originally designed to accept passwords, but rather than specifying the real plaintext password, since users always access Fleet Sandbox from their fleetdm.com account anyway, this generated demo key is used instead to avoid any confusion
         'sandbox_expiration': new Date(fleetSandboxExpiresAt).toISOString(), // sending expiration_timestamp as an ISO string.
       },
       { // Request headers
         'Authorization':sails.config.custom.cloudProvisionerSecret
       }
     )
     .timeout(5000)
-    .intercept('non200Response', ()=>{
-      // If we recieved a non-200 response from the Cloud Provisioner API, we'll throw a 500 error.
-      throw new Error('When attempting to provision a new user\'s Fleet Sandbox instance, the Cloud provisioner gave a non 200 response. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again.');
+    .intercept(['requestFailed', 'non200Response'], (err)=>{
+      // If we recieved a non-200 response from the cloud provisioner API, we'll throw a 500 error.
+      return new Error('When attempting to provision a new user\'s Fleet Sandbox instance, the cloud provisioner gave a non 200 response. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again. Raw response received from provisioner: '+err.stack);
     });
 
-    if(!cloudProvisionerResponse.URL) {
-      // If we didn't receive a URL in the response from the Cloud Provisioner API, we'll throwing an error before we save the new user record and the user will need to try to sign up again.
+    if(!cloudProvisionerResponseData.URL) {
+      // If we didn't receive a URL in the response from the cloud provisioner API, we'll throwing an error before we save the new user record and the user will need to try to sign up again.
       throw new Error(
-        `When provisioning a Fleet Sandbox instance for this new user, the 200 response from the Cloud Provisioner API did not contain a URL of a Sandbox instance.
-        The newly created User record has been deleted, and the user will be asked to try signing up again.
-        Here is the response from the Cloud Provisioner API: ${cloudProvisionerResponse}`
+        `When provisioning a Fleet Sandbox instance for this new user, the response data from the cloud provisioner API was malformed. It did not contain a valid Fleet Sandbox instance URL in its expected "URL" property.
+        The newly created User record , and the user will be asked to try signing up again.
```

Comment:
Looks like some text went missing here

Reply:
I don't think it deleted the User record right?  It's just that no User record was created.